static/js/components/VideoPlayer.js

```diff
@@ -216,6 +228,30 @@ export default class VideoPlayer extends React.Component<*, void> {
     )
   }
 
+  checkYouTube = () => {
```

Comment:
@aliceriot if you have a moment, could you quickly look this function over and tell me if you think there's a better approach?  The purpose is to make sure the video's youtube URL is reachable, and if not modify the VideoJS configuration to play from Cloudfront instead.

Reply:
I think this looks relatively straightforward. The only changes I would suggest are rewriting it using the `async / await` syntax, which I think is generally a little bit more readable. If you did that the function would look something like:

```js
checkYouTube = async () => {
  const { video } = this.props
  if (video.youtube_url) {
    let response
    try {
      response = await fetch(video.youtube_url, {
        mode: "no-cors"
      })
    } finally {
      if (!response || !response.ok) {
        player.reset().src(video.sources).fluid(false)
      }
    }
  }
}
```

or something like that, haha